### PR TITLE
AP_HAL_ChibiOS: don't throw an internal error on mismatched cork/push

### PR DESCRIPTION
**Pull request by andyp1per**
Please decide if you want this or #28825

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/RCOutput.cpp [hunk 1]
```diff
@@ -1329,9 +1329,6 @@ bool RCOutput::get_output_mode_banner(char banner_msg[], uint8_t banner_msg_len)
  */
 void RCOutput::cork(void)
 {
-    if (corked) {
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/RCOutput.cpp:1332]: i'd remove this internal error, but not the other one
calling cork(),cork(),push() isn't great, but is not actually dangerous. Calling push() without cork() is the key issue

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/RCOutput.cpp (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/RCOutput.cpp:1332]: Fixed

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
@IamPete1 I think this is reasonable, but would like you to look too

**Review by IamPete1 — APPROVED** [review]
LGTM

**Comment by timtuxworth** [comment]
Now I'm getting a panic 
https://www.dropbox.com/scl/fi/a6jypd23rnt9dpigamv3q/log_89_2024-12-11-13-44-32.bin?rlkey=bc34pt8or1pa0qy9whw3792ct&st=9ekhb8yx&dl=0

**Pull request merged**